lib/web/fetch/util.js

```diff
@@ -1099,6 +1099,41 @@ function isomorphicEncode (input) {
   return input
 }
 
+/** @type {Uint8ArrayConstructor} */
+const FastBuffer = Buffer[Symbol.species]
+
+/**
+ * Buffer is always cloned and does not use buffer pools.
+ * @param {Uint8Array[]} buffers
+ * @param {number} [length]
+ * @returns {Buffer}
+ */
+function safeBufferConcat (buffers, length) {
```

Comment:
was this taken from node's implementation?

Reply:
I referred to it but it is different.

1. It doesn't check whether the Uint8Array (already checked).
2. It is safe without using pooled buffers and doesn't expose body.

Reply:
wdym by "doesn't expose body"?

Technically this doesn't do anything, the issue only occurs when we don't clone the body and then access .buffer. The spec, for the body mixins, tells us to clone the body. As long as we continue to follow the spec, this won't be an issue, and I'm wondering if taking a performance hit (?) is worth it?

Reply:
> wdym by "doesn't expose body"?

It means that the body is not leaked into the pooled buffer.

Revert #3117 and it seems to have been resolved. But, it can still be reproduced via `Buffer.allocUnsafe(1).buffer.slice(308)`.

Although I don't know, but if NodeJS's security team considers #3337 to be a security threat, it has not actually been resolved.

> I'm wondering if taking a performance hit (?) is worth it?

Basically, they are equivalent.
Slower than current implementation if multiple chunks are smaller than the pool size.

Reply:
If you are concerned about performance, you can do the same optimisation as #3117 again, as `safeBufferConcat` will always generate a new arraybuffer.

Reply:
I am concerned about trading performance for no benefit. We clone the buffer when the spec tells us to which makes this unnecessary. Am I missing something?

Reply:
My general sentiment is spec > performance, but we are already compliant with the spec, from what I see this PR is unnecessarily cloning the buffer that we already clone in the `consumeBody` callback.

Reply:
> My general sentiment is spec > performance

I agreed with you.

> We clone the buffer when the spec tells us to which makes this unnecessary. Am I missing something?

The only difference is that it does not use pooled buffers

It prevents the following:
```js
// node v22.2.0
// import { Response } from "./index.js";

await new Response(new TextEncoder().encode("private-key")).text();

// print private-key
console.log(new TextDecoder().decode(Buffer.allocUnsafe(1).buffer.slice(304)));
```

> I see this PR is unnecessarily cloning the buffer that we already clone in the consumeBody callback.

...I don't have time right now so I will answer later.

Reply:
Would you mind replacing the test you added with that? Much clearer on why we're avoiding allocUnsafe. I'm still not really convinced that it's worth fixing, since it's not a security related bug, but I don't mind this landing.

Reply:
> Would you mind replacing the test you added with that

The test catches this perfectly. What makes you think that way?

Reply:
Ensuring we don't use allocUnsafe is very different from *why* we don't use it.

Reply:
Thanks, I understood what you want to say.